### PR TITLE
daemon: Add hidden --cflags debug command

### PR DESCRIPTION
**Pull request by joestringer**
Support compiling with dedicated compile options to assist development.

### CODE REVIEW

**Review by qmonnet — APPROVED** [review]

**Review by vadorovsky — APPROVED** [review]

**Review by christarazi — APPROVED** [review]

**Review by borkmann — APPROVED** [review]
Do you have a specific example that you typically use?

**Comment by joestringer** [comment]
@borkmann there are a few other flags we could maybe make better use of like `-Wundef`, but in general just having this capability at all is useful to play around with newer clang flags in the container we deploy rather than locally based on my own environment setup.

**Comment by joestringer** [comment]
Given that the flag is hidden, I don't think there's a big deal to wait for the remaining @cilium/cli codeowner review (especially given  there's already four reviews). Good to merge.

**Pull request merged**
